server.js

```diff
@@ -198,28 +198,33 @@ function Server (opts) {
           if (keys.length > 0) activeTorrents++
 
           keys.forEach(function (peerId) {
-            if (!allPeers.hasOwnProperty(peerId)) {
-              allPeers[peerId] = {
-                ipv4: false,
-                ipv6: false,
-                seeder: false,
-                leecher: false
-              }
-            }
             // Don't mark the peer as most recently used for stats
             var peer = peers.peek(peerId)
-            if (peer.ip.indexOf(':') >= 0) {
-              allPeers[peerId].ipv6 = true
-            } else {
-              allPeers[peerId].ipv4 = true
-            }
-            if (peer.complete) {
-              allPeers[peerId].seeder = true
-            } else {
-              allPeers[peerId].leecher = true
+
+            // The peer could be evicted at this point
+            if (typeof peer !== 'undefined') {
```

Comment:
It's usually better to invert if-statements like this one to prevent another level of indentation.

This is better, IMO:

```js
if (peer == null) return
```

Also, it's better to be defensive and treat `null` like no peer as well, hence the `==`.